src/main/java/com/ea/restaurant/util/GrpcUtil.java

```diff
@@ -0,0 +1,43 @@
+package com.ea.restaurant.util;
+
+import com.ea.restaurant.constants.RequestMetadataConstants;
+import com.ea.restaurant.exceptions.UnauthorizedException;
+import com.ea.restaurant.grpc.interceptors.GrpcServerInterceptor;
+import com.ea.restaurant.record.Oauth2ClientCredentials;
+import io.grpc.Metadata;
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+import java.util.Optional;
+import java.util.stream.Stream;
+
+public class GrpcUtil {
+
+  public static Oauth2ClientCredentials getCredentialsFromBasicAuthToken(Metadata metadata) {
+    var authorization =
+        metadata.get(
+            Metadata.Key.of(
+                RequestMetadataConstants.KEY_NAME_FOR_AUTHORIZATION_METADATA,
+                Metadata.ASCII_STRING_MARSHALLER));
+
+    var basicToken = extractBasicToken(authorization);
+    return getBasicAuthenticationDecoded(basicToken);
+  }
+
+  private static String extractBasicToken(String authorization) {
```

Comment:
This should be public. What if I want to use this in another place?

Reply:
Add tests to these public methods

Reply:
Done.